File: classfields/test/java/ru/auto/ara/RobolectricTest.java
package ru.auto.ara;

import android.app.Application;

import com.yandex.mobile.verticalcore.utils.AppHelper;

import org.junit.Before;
import org.junit.runner.RunWith;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

import ru.auto.test.runner.AllureRobolectricRunner;

/**
 * Base test setup for running with robolectric + PowerMock test runner
 * Created by aleien on 26.12.16.
 */

@RunWith(AllureRobolectricRunner.class)
@Config(sdk = 23, packageName = "ru.auto.ara.debug", manifest = Config.NONE, application = AutoApplication.class)
public abstract class RobolectricTest extends RxTest {
    private boolean isInitialSetup = true;

    @Before
    public void setupPowerMockito() {
        if (isInitialSetup) {
            // Setup for AppHelper (it's being overriden by PowerMockRule)
            Application app = RuntimeEnvironment.application;
            AppHelper.setupApp(app);
            AutoApplication.mainComponentHolder.initialize();

            isInitialSetup = false;
        }
    }

}

File: classfields/test/java/com/yandex/mobile/realty/RobolectricTest.kt
package com.yandex.mobile.realty

import android.app.Application
import android.content.Context
import androidx.appcompat.view.ContextThemeWrapper
import androidx.test.core.app.ApplicationProvider
import com.yandex.mobile.verticalcore.utils.AppHelper
import org.junit.Before
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

/**
 * @author rogovalex on 19/02/2019.
 */
@RunWith(RobolectricTestRunner::class)
@Config(
    sdk = [21],
    packageName = BuildConfig.APPLICATION_ID,
    manifest = Config.NONE,
    application = TestApplication::class,
    qualifiers = "ru"
)
abstract class RobolectricTest {

    protected lateinit var context: Context

    @Before
    fun baseSetup() {
        val application = ApplicationProvider.getApplicationContext<Application>()
        AppHelper.setupApp(application)
        context = ContextThemeWrapper(application, R.style.AppTheme)
    }
}
